app/models/deploy_baseline.rb

```diff
@@ -0,0 +1,35 @@
+# frozen_string_literal: true
+class DeployBaseline
+  def initialize(deploy)
+    @deploy = deploy
+  end
+
+  def total_latency
+    @total_latency ||= begin
+      {
+        pr_production_latency: valid_deployment ? pr_production_latency : 0,
+        staging_production_latency: valid_deployment ? staging_production_latency : 0
+      }
+    end
+  end
+
+  private
+
+  def valid_deployment
+    @deploy&.production && @deploy&.status == "succeeded"
+  end
+
+  def pr_production_latency
+    pull_requests = @deploy&.changeset&.pull_requests || []
+    pull_requests.sum { |pr| @deploy.updated_at.to_i - pr.created_at.to_i }
+  end
+
+  def staging_production_latency
+    jobs = Job.where(commit: @deploy&.commit)
+    stages = Stage.where(project: @deploy&.project)
+    stages = stages.reject(&:production?)
+
+    first_staging_deploy = Deploy.where(job: jobs, stage: stages).reorder(:id).first
+    first_staging_deploy ? @deploy.updated_at.to_i - first_staging_deploy.created_at.to_i : 0
```

Comment:
again,. needs explanation on what this means exactly, but I was thinking if it's time it takes from staging to prod then we should measure "staging done to production done" 

```suggestion
    first_staging_deploy ? @deploy.updated_at.to_i - first_staging_deploy.updated_at.to_i : 0
```